### DISCUSSION TITLE
Auto-import takes priority over local variable(s) when it is a full match in autocomplete

### DISCUSSION THREAD

**Discussion by fireattack** [microsoft/pylance-release#3157] [Enhancement]
## Environment data

-   Language Server version: 2022.8.10 (pyright 6fa755c0)
-   OS and version: Windows 7
-   Python version (& distribution if applicable, e.g. Anaconda): 3.8.7

## Code Snippet

```python
def parse_page(url):
    print('Dummy')

```

## Repro Steps

1. create a .py file with the above code in it.
2. At the bottom, start typing "parse".

## Expected behavior

"parse_page" should be the first priority all the time.

## Actual behavior

It is the first priority when you have typied p, pa, par, and pars. But once you typed the whole "parse", it will start to show a long list of auto-import, and the desired function will be at the bottom of the list. If you type "parse_", it then immediately begins to become the first again. 

Please see this screencap to get better idea:
![quick12](https://user-images.githubusercontent.com/4130991/183430555-1a8245fb-4f6e-40de-8610-6ec526c3c67f.gif)

## Comment

I personally think this behavior is very strange. To autocomplete a local variable I have to carefully to not type too much, otherwise after tabbing (muscle memory) it will not only gives me the wrong thing, but also adds something I don't need to `import`. 

And the fact that autocomplete results could change dramatically before and after typing one more letter seems undesirable from the UX perspective.

**Comment by judej** [comment]
Looking for feedback from users on the preference of local over exact matches from imported modules.

Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by fireattack** [comment]
Another case is when importing from a local sibling script:

![quick12](https://user-images.githubusercontent.com/4130991/184260608-7dd670b0-aaa4-4178-9869-1a9a43d5dcf6.gif)

As you can see here, I have an function called "format_str" in the util.py. When I tried to import it, it only lists it as the first choice when I typed "f" or "fo"; afterwards, it was deprioritized for no reason. This makes even less sense considering the context is already `from util`.

I'm pretty sure this problem (along with the one in OP) didn't exist before. I assume something changed?

**Comment by pukhov** [comment]
Indeed, very annoying -- you start typing function name, e.g. "print", on "pr" it shows print, you want to press tab, but you already typed "pri" at this moment and it autocompletes with another function.
